toolkit/tools/internal/pkggraph/pkggraph_test.go

```diff
@@ -600,11 +620,20 @@ func TestConditionalLookupBasic(t *testing.T) {
 	assert.NotNil(t, lu)
 	assert.True(t, lu.RunNode.Equal(pkgCRun))
 
-	// Best is D2, which is ver: <=2
+	// Best is D2, which is ver: <=2, BUT since we can't predict the future its not safe to use it
+	lu, err = g.FindBestPkgNode(&pkgjson.PackageVer{Name: "D", Version: "1"})
+	assert.NoError(t, err)
+	assert.Nil(t, lu)
+
+	// If we add a new special exact version of D, we should get that one
+	pkgDExactly1 := pkgjson.PackageVer{Name: "D", Version: "1", Condition: "="}
+	pkgpkgDExactly1Unresolved := buildUnresolvedNodeHelper(&pkgDExactly1)
```

Comment:
To be consistent with how this test is written, I think we should make this new node part of `allNodes`. Despite how I dislike how this test is written in general.:)